### PR TITLE
added health bar and internal game timers

### PR DESCRIPTION
**Pull request by lourw**
# Description

* added a health bar into the game that listens to an internal game timer
* added internal game timer that will send listeners events when it is time to update
  * hardcoded depletion of virtual pet health currently 1 point per second

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added tests for health bar and health timer within the game. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wu-jacob — APPROVED** [review]
Looks good. I like the death animation lol

File: front-end/virtual_pet/lib/flame/game/vitural_pet_game.dart [hunk 1]
```diff
@@ -1,12 +1,20 @@
 import 'package:flame/game.dart';
```
Comment (mebesius) [inline front-end/virtual_pet/lib/flame/game/vitural_pet_game.dart]: Can we rename this file so it is `virtual_pet_game.dart` and not `vitural_pet_game.dart`?

**Review by mebesius — CHANGES_REQUESTED** [review]

File: front-end/virtual_pet/lib/flame/game/vitural_pet_game.dart (hunk 1 above)
Comment (mebesius) [reply front-end/virtual_pet/lib/flame/game/vitural_pet_game.dart]: Really nice otherwise, very clean code 👍

Comment (lourw) [reply front-end/virtual_pet/lib/flame/game/vitural_pet_game.dart]: Good catch!

**Review by mebesius — APPROVED** [review]
Thanks for changing that!

**Pull request merged**